simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/handlers/ocpp/CancelReservationRequestHandler.java

```diff
@@ -22,37 +23,27 @@
     @Override
     public void handle(String callId, CancelReservationRequest request) {
         Optional<Reservation> reservationOpt = stationStore.tryFindReservationById(request.getReservationId());
+        CancelReservationResponse cancelReservationResponse = reservationOpt.map(reservation -> new CancelReservationResponse().withStatus(CancelReservationResponse.Status.ACCEPTED))
+                .orElseGet(() -> new CancelReservationResponse().withStatus(CancelReservationResponse.Status.REJECTED));
 
-        if (!reservationOpt.isPresent()) {
-            stationMessageSender.sendCallResult(callId, new CancelReservationResponse().withStatus(CancelReservationResponse.Status.REJECTED));
-        } else {
-            Reservation reservation = reservationOpt.get();
-            stationMessageSender.sendCallResult(callId, new CancelReservationResponse().withStatus(CancelReservationResponse.Status.ACCEPTED));
-
-            if (isConnectorReserved(reservation)) {
+        reservationOpt.ifPresent(reservation -> {
+            if (reservation.getEvse().getConnectorId() != null && isConnectorReserved(reservation)) {
```

Comment:
lets use more of Optionals
```
reservationOpt
.filter(reservation.getEvse().getConnectorId() != null && isConnectorReserved(reservation))
.ifPresent(...)
```

internals of filter  and ifPresent could be extracted to a function and unit-tested in isolation

Reply:
Actually, I made a mistake. I need to remove reservation if it was found and separate from that to make connector available if it is reserved. Can you check again?